### ISSUE TITLE
Synchronize data with other fitness trackers

### ISSUE THREAD

**Issue by rolandgeider** [wger-project/wger#1111]

**Comment by cvantum** [comment]
Hello,

i saw your project at the hacktoberfest 2022. Related to this issue i would prospose to implement Strava since most fitness trackers (like Garmin, Wahoo, Polar) build already a connection. So tracked Data can be synced with strava and should be available via API (https://developers.strava.com/docs/reference/#api-Activities-getActivityById).

An actictiy has also a keyword for `calories`. The only downside could maybe the API-Access Limitation:

```
Strava API usage is limited on a per-application basis using both a 15-minute and daily request limit. The default rate limit allows 100 requests every 15 minutes, with up to 1,000 requests per day.
```

**Comment by rolandgeider** [comment]
Hi! From what I've seen, strava does mainly things like running, cycling and so on, right? Also the calories we get are the ones consumed during the activity which we currently don't track (but could, this wouldn't be a bad idea, I've opened a new issue for this)

**Comment by MrRulf** [comment]
I think gadgetbridge can also be used as a provider for data from smart devices. FitoTrack also seems like an option. Just to provide compatibility with open source watches.

**Comment by fuveluck** [comment]
Hi threre, Can I ask what situation is with this issue? Is there any progress in discussion?

**Comment by rolandgeider** [comment]
The idea is to add some snychronization service that periodically pulls datainto our database. Things that we still need to do/find out:

* which services offer some kind of api, and which data they can sync. 
* select one popular one to start with
* add the necessary UI (how is the user authenticated? access token? OAuth? do we have to register the app somewhere? etc)
* implement a function that does the sync for the one service. Probably something like body weight data is the easiest to begin with, but can be something else
* if necessary, make needed changes to the wger database. E.g. it would be perfectly ok if we would add a new model to track the spent calories and this could only be filled via these imports, and not directly over the app itself.
* add the function to the celery queue so that it can run periodically

So basically the specifics on this depend on where we get the data from. If you are interested @fuveluck I can take a look and see which options we have on that side

**Comment by CEbbinghaus** [comment]
Pretty sure that Google Fitness has a pretty good API encouraging others to integrate with it. As such I would propose that as a first integration, Google would be used.

**Comment by MrRulf** [comment]
Android 14 introduced Health Connect, completely on the device and as part of AOSP, not play services exclusive or something, I would highly recommend to make compatibility with that the priority, before custom solutions for other apps get built, as the idea behind Health Connect seems to be that it becomes the main place, where all health data can be synced to and exchanged over it trough different apps

**Comment by rolandgeider** [comment]
yes, somebody at Fosdem also suggested this, sounds like something we want to investigate further. A (very) quick search resulted in two flutter packages that seem to support this, so this might be relatively painless to do (well, besides all the backend changes we will need 😅 )

**Comment by MrRulf** [comment]
Oof, hope its not too much stuff ^^"
At least helth connect seems like something that is going to stay, considering foss people will prob like it cuz its not bound to google, samsung or anyone, while the projects at the same time still have google, samsung and i think others too behind it ^^

**Comment by rolandgeider** [comment]
well, currently there is no good place to store such data. Plus we will want to show it somewhere useful. But if the import is easy, that will make a big difference, I was worried we would need to maintain custom code for each service

**Comment by MrRulf** [comment]
It seems lile this could solve the custom code for each service problem quite well ^^
I dont know if health connect has built in methods to visualize data, I think it really is mainly a llace to collect and distribute the data, so the visualizing can be done by different apps, wger can say they wanna build that into the app directly, but users could also just have another app to visualize the data. I really like how modular it is and that apps now dont have to bother with fancy visualisation or processing when they just track one simple exercise, while apps with focus on processing and visualisation, maybe even without the ability to track excersises, can exist now and do what they are great at on all the health data a user provides

**Comment by CEbbinghaus** [comment]
Health Connect sounds like a far superior option. Being bound to Android 14 isn't perfect as there are plenty of devices in use today that will never get the new version but Looking into the future it seems like the best choice. And keeping all data on device while allowing all kinds of fitness apps to communicate data is perfect from a privacy standpoint. I wholeheartedly support this proposal

**Comment by Dieterbe** [comment]
Strava has not shown interest in syncing their data to health connect (see [this support thread](https://communityhub.strava.com/t5/devices-and-connections/health-connect-android/m-p/17011)). However, they do seem to have some sync with Apple Health ([they have a tutorial for it](https://support.strava.com/hc/en-us/articles/216917527-Health-App-and-Strava#:~:text=Syncing%20from%20Strava%20to%20Health&text=Strava%20will%20send%20data%20to,will%20not%20sync%20to%20Health.)) so perhaps health connect will follow too..
